### ISSUE TITLE
Amazon EKS AMI creates containerd socket at diferent path

### ISSUE THREAD

**Issue by rafaelgaspar** [DataDog/helm-charts#338]
**Describe what happened:**

Amazon EKS AMI is currently creating the containerd socket at `/run/dockershim.sock`, the reason for that is that `aws-cni-plugin` still points to that path, so they cannot at the moment just change to `/run/containerd/containerd.sock`.

Setting `criSocketPath: /run/dockershim.sock` works for CRI metrics, but not for containerd metrics.

**Describe what you expected:**

containerd metrics to be available, maybe have a way to set a different mountPath for the criSocketPath inside datadog container so that it recognises it is a containerd socket.

**Steps to reproduce the issue:**

Spin up cluster on EKS and set `--container-runtime containerd` for `bootstrap.sh`

**Additional environment details (Operating System, Cloud provider, etc):**

EKS AMI v1.21.2-13+d2965f0db10712

**Comment by rafaelgaspar** [comment]
BTW, there is also https://github.com/awslabs/amazon-eks-ami/pull/724

**Comment by cw-sakamoto** [comment]
After making the following settings, it worked even if `dockershim.sock` for containerd's cri path.

```
agents:
  volumes:
    - hostPath:
        path: /var/run/dockershim.sock
      name: crisocketpath
  volumeMounts:
    - name: crisocketpath
      mountPath: /var/run/containerd/containerd.sock
      readOnly: true
 ```

**Comment by bubalush** [comment]
> After making the following settings, it worked even if `dockershim.sock` for containerd's cri path.
> 
> ```
> agents:
>   volumes:
>     - hostPath:
>         path: /var/run/dockershim.sock
>       name: crisocketpath
>   volumeMounts:
>     - name: crisocketpath
>       mountPath: /var/run/containerd/containerd.sock
>       readOnly: true
> ```

What about default `kubernetes.*` metrics of the pods and etc.?

**Comment by vboulineau** [comment]
Hello,

Setting `criSocketPath: /run/dockershim.sock` is indeed the right thing to do, however you don't get `containerd` automatically activated simply because the auto-detection (which is based on socket name) fails to recognize this as a `containerd` socket.

The proper way to force activate a feature that's not autodetected is to use:
`autoconfig_include_features: ["containerd"]` or `DD_ AUTOCONFIG_INCLUDE_FEATURES=containerd`

In the Helm chart, this can be set through a custom `datadog.yaml` or through `datadog.env` in `values.yaml`

This is available since Agent 7.29. We will update the [Kubernetes distribution documentation](https://docs.datadoghq.com/agent/kubernetes/distributions/?tab=helm#EKS) with this information.

**Comment by saiindrakaran** [comment]
I also face similar issue as @cw-sakamoto  , although my agent status looks [OK], I can't confirm metrics for kubenetes.{cpu,memory,filesystem,network}.*.

```
containerd
    ----------
      Instance ID: containerd [OK]
      Configuration Source: file:/etc/datadog-agent/conf.d/containerd.d/conf.yaml.default
      Total Runs: 16
      Metric Samples: Last Run: 363, Total: 5,808
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 1, Total: 16
      Average Execution Time : 323ms
      Last Execution Date : 2021-08-28 00:57:15 UTC (1630112235000)
      Last Successful Execution Date : 2021-08-28 00:57:15 UTC (1630112235000)

kubelet (7.0.0)
    ---------------
      Instance ID: kubelet:5bbc63f3938c02f4 [OK]
      Configuration Source: file:/etc/datadog-agent/conf.d/kubelet.d/conf.yaml.default
      Total Runs: 12
      Metric Samples: Last Run: 754, Total: 9,048
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 4, Total: 48
      Average Execution Time : 992ms
      Last Execution Date : 2021-08-28 00:57:04 UTC (1630112224000)
      Last Successful Execution Date : 2021-08-28 00:57:04 UTC (1630112224000)
      


```
